solutions/devsprint-michel-ferreira-1/src/main/kotlin/com/devpass/challengehexagonal/domain/entity/TransactionEntity.kt

```diff
@@ -0,0 +1,22 @@
+package com.devpass.challengehexagonal.domain.entity
+
+import java.math.BigDecimal
+import java.time.LocalDateTime
+import javax.persistence.*
+
+@Entity
+data class TransactionEntity(
+    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
+    val id: Long? = null,
+
+    val establishment: String,
+
+    val transactionDate: LocalDateTime,
+
+    val value: BigDecimal,
+
+    val account: AccountEntity,
+
+//    @ManyToOne(optional = false)
+//    @JoinColumn(name = "account_id", referencedColumnName = "account")

```

Comment:
Tivemos dúvidas para usar as anotações @OneToMany/https://github.com/manytoone nessa parte, pois sempre aparece um erro (This annotation is not applicable to target 'value parameter'), vc poderia dar um help, @michel05 ? 😄

Reply:
Coloquei alguns comentários no PR para representar como ficaria.
Mas de forma resumida esse nosso cenário usa uma abordagem bidirecional, ou seja, temos as duas pontas da relação representadas nas nossa entidades.
Há algumas particularidades e cuidados ao usarmos elas.

Em um modelo biderional a melhor forma é tratarmos os dois lados com fetch = FetchType.LAZY, para evitar um stackoverflow.
No nosso OneToMany representar com o mappedBy qual o nome da tabela referente.
No nosso ManyToOne, apenas adicionar o fetch = FetchType.LAZY
O JoinColumn pode ser um auxiliar, mas no modelo biderecional ele não é necessário para nosso cenário.

Reply:
Ahhhh, agora fez muito mais sentido, obrigada Michel! Corrigi com base nos apontamentos!